tests/tasks/setup/kit/dataplane.yaml

```diff
@@ -0,0 +1,101 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: data-plane-setup
+  namespace: tekton-pipelines
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/categories: Kubernetes
+    tekton.dev/tags: CLI, kubectl
+    tekton.dev/displayName: "kubernetes actions"
+    tekton.dev/platforms: "linux/amd64"
+spec:
+  description: |
+    Setup a kubernetes data plane in the guest cluster.
+  params:
+  - name: cluster-name
+    default: "guest"
+    description: Name of the guest cluster
+  - name: dataplane-name
+    default: ""
+    description: Optional name of the dataplane object.
+  - name: desired-nodes
+    default: "10"
+    description: The desired number of nodes in the cluster.
+  - name: min-nodes
+    default: "1"
+    description: The minimum number of nodes in the cluster nodegroup.
+  - name: max-nodes
+    default: "1000"
+    description: The maximum number of nodes in the cluster nodegroup.
+  - name: instance-types
+    default: '["c5.large","m5.large","r5.large","t3.large","t3a.large","c5a.large","m5a.large","r5a.large"]'
+    description: The desired instance types of the node group.
+  workspaces:
+  - name: config
+    mountPath: /config/
+  steps:
+  - name: setup-data-plane
+    image: bitnami/kubectl:1.24.5
+    script: |
+      #!/bin/bash
+      set -e
+      echo "Getting subnet and security group tags"
+      TAG=$(kubectl get provisioner -oyaml | grep karpenter.sh/discovery | awk 'NR==1{ print $2}')
+      echo "Setting up data plane"
+      max_nodes=$(params.max-nodes)
+      nodes=$(params.desired-nodes)
+      asgs=$((nodes/max_nodes))
+      echo "node groups: $asgs"
+      if [ -z "$(params.dataplane-name)" ]
+      then
+        dp_name=$(params.cluster-name)-nodes
+      else
+        dp_name=$(params.dataplane-name)
+      fi
+
+      create_dp_nodes()
+      {
+        dataplane_name=$dp_name-$1
+        EC2_INSTANCES=$3
+        cat <<EOF >> /tmp/dp.yaml
+      apiVersion: kit.k8s.sh/v1alpha1
```

Comment:
let's not embed these things and give flexibility to user to pass in as file location be it github or s3 whatever makes most sense.